pkg/printer/printer.go

```diff
@@ -0,0 +1,93 @@
+package printer
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/landoop/tableprinter"
+	"github.com/yalp/jsonpath"
+	"os"
+)
+
+type Printer struct{}
+
+func (p Printer) PrintOutput(output string, i interface{}) {
+	// Factory Method for all printer
+	fmt.Println("==",output)
+	switch output {
+	case "json": // Print protobuf to json
+		result, err := json.Marshal(i)
+		if err != nil {
+			os.Exit(1)
+		}
+		fmt.Println(string(result))
+		break
+	case "yaml": // Print protobuf to yaml
+	    
+		break
+	default: // Print table
+
+		printer := tableprinter.New(os.Stdout)
+		printer.Print(i)
+		break
+	}
+}
+
+
+func(p Printer) BuildOutput(input []interface{},column map[string]string,resultType string) ([]interface{},error) {
+	response := make([]interface{}, 0, len(input))
+	for _, data := range input {
+
+		tableData := make(map[string]interface{})
+		for k := range column {
+			data, _ := jsonpath.Read(data, column[k])
+			tableData[k] = data.(string)
+		}
+		jsonbody, err := json.Marshal(tableData)
+		if err != nil {
+			return response,err
+		}
+		switch resultType {
+			case "PrintableTask":
+				results := PrintableTask{}
+				if err := json.Unmarshal(jsonbody, &results); err != nil {
+					return response,err
+				}
+				response = append(response, results)
+				break;
+			case "PrintableWorkflow":
+				results := PrintableWorkflow{}
+				if err := json.Unmarshal(jsonbody, &results); err != nil {
+					return response,err
+				}
+				response = append(response, results)
+				break;
+			case "PrintableProject":
+				results := PrintableProject{}
+				if err := json.Unmarshal(jsonbody, &results); err != nil {
+					return response,err
+				}
+				response = append(response, results)
+				break;
+			case "PrintableNamedEntityIdentifier":
```

Comment:
I think I am missing something,
I see that you have create a special structure to print a table, but that is the only difference between all "cases"
I want to think of a way of getting rid of the structure for printing a table. That way, all of this code will be once and then we can just start adding all command

@EngHabu you have more context about this library - comments?